16_threading_multiprocessing.py

```diff
@@ -0,0 +1,39 @@
+import requests
+import threading
+import time
+
+cities = [
+    {"name": "Calgary", "latitude": 51.05, "longitude": -114.09},
+    {"name": "Da Nang", "latitude": 16.07, "longitude": 108.22},
+    {"name": "Nanchang", "latitude": 28.68, "longitude": 115.85},
+    {"name": "Kyiv", "latitude": 50.45, "longitude": 30.52},
+    {"name": "Ivano-Frankivsk", "latitude": 48.92, "longitude": 24.71}
+]
+
+def get_weather():
+    url = f"https://api.open-meteo.com/v1/forecast?latitude={'latitude'}&longitude={'longitude'}&hourly=temperature_2m"
+    response = requests.get(url)
+    data = response.json()
+    temperature_list = data["hourly"]["temperature_2m"]
+    avg_temperature = sum(temperature_list) / len(temperature_list)
+    return (city, avg_temperature)
+
+def get_average_temperatures():
+    temperatures = []
+    with threading.Lock():
```

Comment:
there is no reason of using Lock

Reply:
@acman , I tried to edit it as much as I could. [https://github.com/AStefankiv/Stefankiv-Homework/blob/HW_16/16_threading_corrected.py](url)

However, I couldn't make one code. I made 2. One with threading, another with multiprocessing. Please take a look